pennylane/ops/op_math/prod.py

```diff
@@ -84,7 +85,26 @@ def prod(*ops, do_queue=True, id=None, lazy=True):
     >>> prod_op.matrix()
     array([[ 0, -1],
            [ 1,  0]])
+
+    You can also create a prod operator by passing a qfunc to prod, like the following:
+
+    >>> def qfunc(x):
+    ...     qml.RX(x, 0)
+    ...     qml.CNOT([0, 1])
+    >>> prod_op = prod(qfunc)(1.1)
+    >>> prod_op
+    CNOT(wires=[0, 1]) @ RX(1.1, wires=[0])
     """
+    if len(ops) == 1 and callable(ops[0]):
+        fn = ops[0]
+
+        @wraps(fn)
+        def wrapper(*args, **kwargs):
+            qs = qml.tape.make_qscript(fn)(*args, **kwargs)
```

Comment:
What behavior do we want if the qfunc only contains one object?

Since this will essentially be a general purpose qfunc to operator converter, I think its a valid situation that will occur. I'd be happy with just returning that individual op.

Though this does make me wonder if we should have some standalone function other than `qml.prod` as an "qfunc to operator" converter, since `qml.prod` means "product of multiple operators".

How do we *define* the error from a single operator qfunc out of existence?

Reply:
imo it currently fails as expected (saying that a prod must have more than one arg). I agree that if we want to support it, that probably belongs in a standalone function

Reply:
> > Bringing in @trbromley for this discussion. What do we think about the syntax:
> > ```
> > qml.prod(qfunc)(*args, **kwargs)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Yes in this case we will return a function instead of a `Prod` operator, but we do support this behaviour in other places already, like `adjoint`.
> > If we don't want to have the same function with different arguments and outputs, maybe we could just create a new function `qml.qfunc_to_opeator`, or something else better named?
> 
> +1 to having `qml.prod()` accept a `qfunc` and return another callable that returns a `Prod` when called.
> 
> @DSGuala @Jaybsoni will this help solve the issue with using `qfunc`s in `qml.QSVT`?

This will definitely make it easier to define custom `BlockEncoding` and `PCPhase` operations in `qml.QSVT`. I gave it a quick try and found it helpful :+1: 

Using `qml.prod` is still an extra step over submitting qfuncs directly to `qml.QSVT`. But it's much better than having to create a loop using `qml.prod` to get all the operations into one `qml.prod`.

We could potentially have `qml.QSVT` check the type the operations submitted to it and if they are qfuncs, automatically use `qml.prod` under the hood.